web/impact/impact/utils.py

```diff
@@ -1,15 +1,64 @@
 # MIT License
 # Copyright (c) 2017 MassChallenge, Inc.
+from impact.models import BaseProfile
+
+REQUIRED_USER_KEYS = [
+    "email",
+    "first_name",
+    "last_name",
+]
+OPTIONAL_USER_KEYS = [
+    "is_active",
+]
+USER_KEYS = REQUIRED_USER_KEYS + OPTIONAL_USER_KEYS
+REQUIRED_PROFILE_KEYS = [
+    "gender",
+]
+PROFILE_KEYS = REQUIRED_PROFILE_KEYS
+ALL_USER_RELATED_KEYS = USER_KEYS + PROFILE_KEYS
+KEY_TRANSLATIONS = {
+    "first_name": "full_name",
+    "last_name": "short_name",  # Yes, this is correct
+}
+VALID_GENDERS = ["f", "m", "o", "p"]
+INVALID_GENDER_ERROR = ("Invalid gender: '{}'. Valid values are "
+                        "'f' or 'female', 'm' or 'male', "
+                        "'o' or 'other', and 'p' or 'prefer not to state'")
+GENDER_TRANSLATIONS = {
+    "female": "f",
+    "male": "m",
+    "other": "o",
+    "prefer not to state": "p",
+}
 
 
 def compose_filter(key_pieces, value):
     return {"__".join(key_pieces): value}
 
 
 def get_profile(user):
-    user_type = user.baseprofile.user_type
-    if user_type == "ENTREPRENEUR":
-        return user.entrepreneurprofile
-    if user_type == "EXPERT":
-        return user.expertprofile
-    return user.memberprofile
+    try:
+        user_type = user.baseprofile.user_type
+        if user_type == "ENTREPRENEUR":
+            return user.entrepreneurprofile
+        if user_type == "EXPERT":
+            return user.expertprofile
+        return user.memberprofile
+    except BaseProfile.DoesNotExist:
+        return None
+
+
+def user_gender(user):
+    profile = get_profile(user)
+    if profile:
+        return profile.gender
+    return None
+
+
+def find_gender(gender):
+    if not hasattr(gender, "lower"):
+        return None
+    gender = GENDER_TRANSLATIONS.get(gender.lower(), gender)
+    if gender in VALID_GENDERS:
+        return gender
+    return None
```

Comment:
This is essentially handled by the choices on the gender field of CoreProfile which essentially imposes the same constraint on drop down UI elements representing gender.  The challenge with the API of course is we don't have a UI element to do this work for us.